### ISSUE TITLE
JuMP v0.21 Support

### ISSUE THREAD

**Issue by ccoffrin** [lanl-ansi/Juniper.jl#176]
The following example,
```
using Juniper
```
seems to result in the following error,
```
ERROR: LoadError: LoadError: UndefVarError: OptimizerFactory not defined
```

**Comment by ccoffrin** [comment]
Started this in the `jump-v0.21` branch.  `src/bb_inits_and_defaults.jl` needs to be updated to support the new `OptimizerWithAttributes` feature, otherwise seems to be working.  I'll leave that to you are your earliest convenience.

**Comment by Wikunia** [comment]
Implemented by #179
